docs/user_guide/examples.md

```diff
@@ -1,14 +1,248 @@
+The sandbox can be defined in the task class itself using the syntax shown above, elsewise the default sandbox defined in the analysis config file is used.
+
+
+### The parameters:
+
+Like the arguments of a function, the parameters of a task are the variables that are passed to the task when it is run from the command line (``law run MyTask --param1 foo``), making the task very flexible.
+The parameters are defined as class attributes and are identified as such by the class method `get_params()`.
+The parameters returned by this method are parsed from the command line by the class method `get_param_values(params, args, kwargs)`.
+If a parameter is not provided in the command line, the default value is used.
+Finally, the parameters are initialized in the `__init__()` method of the task with the task object itself and can be accessed across all methods of its instance.
+
+The parameters are defined using the [luigi](https://luigi.readthedocs.io/en/stable/api/luigi.parameter.html) or [law](https://law.readthedocs.io/en/latest/api/parameter.html) parameters, which offer a wide range of different parameter types with many helpful options.
+One will often encounter the base {external+law:py:class}`~law.parameter.Parameter`, which parses strings.
+Moreover, parameters from [luigi](https://luigi.readthedocs.io/en/stable/api/luigi.parameter.html) are equally useful, when defining parameters that parse integers, dates, lists, and so on.
+If needed, one can also define custom parameters.
+However, it is important to inherit the parameter class from the package used for the task (luigi or law).
+Otherwise, the parameter will not work correctly.
+
+### The requirements:
+
+#TODO explain branches if not explained in law
+
+As noticed in the [code snipped](#example-task-class), a task can define three different requirement keywords, which have different functionality in the framework.
+
+The ```requires()``` methode in a task, which is inherited from {external+law:py:class}`law.task.base.Task`, defines all required outputs for a task to run on a branch level.
+If an output does not exist, the task responsible for this output will be scheduled and executed before the desired task is run.
+
+The {external+law:py:meth}`~law.tasks.ForestMerge.workflow_requires()` is analog to the former methode, however it defines the dependencies on a workflow level, which can be very useful, when working remotely and running jobs on remote clusters.
+Every task defines a workflow, which contains all available branches of this task.
+When executing the workflow, the task will run in all (specified) branches.
+If the desired workflow requires the output of another workflow, the latter will be executed beforehand, which means that all needed branches (with no existing output) from the required workflow will be executed before any of the branches from the current workflow runs (Workflow runs vertically).
```

Comment:
Maybe some sort of illustration to show what `runs vertically` or `runs horizontally` means would be helpful here

Reply:
The `req plot`, which was not rendered correctly is the illustration for this :)